### PR TITLE
For #18376: Add experiment for set default browser settings card.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

Screenshots here: https://github.com/mozilla-mobile/fenix/issues/18376#issuecomment-818903459

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt [hunk 1]
```diff
@@ -136,11 +139,21 @@ class SettingsFragment : PreferenceFragmentCompat() {
...
 
     override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
-        val preferencesId = if (FeatureFlags.newIconSet) {
-            R.xml.preferences_without_icons
+
+        val preferencesId = if (isExperimentBranch()) {
+            if (FeatureFlags.newIconSet) {
+                R.xml.preferences_without_icons_experiment
+            } else {
+                R.xml.preferences_experiment
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:147]: As this `XML` belongs to a specific experiment may we should add the name of the experiment to identify it

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt [hunk 2]
```diff
@@ -578,6 +591,13 @@ class SettingsFragment : PreferenceFragmentCompat() {
         }
     }
 
+    fun isExperimentBranch(): Boolean {
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:594]: nit: Maybe `isDefaultBrowserExperimentBranch`

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt (hunk 1 above)
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:147]: Having the extra `XML` files make modification a bit harder as we will have to make change in 4 files, but I can't think on better way to do it, without making things a bit complex, and as these changes will be momentarily I guess it's OK. Maybe we could extract the getting the preference ID into a function and add some comments for people to be aware that changes to the setting `XML` could required changes on the 4 `XML` files.

**Review by Amejia481 — CHANGES_REQUESTED** [review]
It's looking good, I added some commends 👍🏽 .

There are some functionality still missing maybe they will added later:
* The default browser item should be re-styled and be the first item [see](https://www.figma.com/file/w8vFnZ05xMgDPkEb9r0r35/R89?node-id=1%3A1720).
* The sync menu item should be re-styled [see](https://www.figma.com/file/w8vFnZ05xMgDPkEb9r0r35/R89?node-id=12%3A3297). 
*  For users with Firefox as default browser we should present the setting pages without any changes [see](https://www.figma.com/file/w8vFnZ05xMgDPkEb9r0r35/R89?node-id=1%3A1877), as it's on master now.

**Comment by Amejia481** [comment]
Additionally the experiment shouldn't show for people that have Nightly or Beta as their default browsers.

**Comment by Amejia481** [comment]
> Additionally the experiment shouldn't show for people that have Nightly or Beta as their default browsers.

For this we can use [browsers.isFirefoxDefaultBrowser](https://github.com/mozilla-mobile/android-components/blob/d4307a6917913bf5e5776b0b96f1929dd645653c/components/support/utils/src/main/java/mozilla/components/support/utils/Browsers.kt#L125).

**Comment by mcarare** [comment]
> > Additionally the experiment shouldn't show for people that have Nightly or Beta as their default browsers.
> 
> For this we can use [browsers.isFirefoxDefaultBrowser](https://github.com/mozilla-mobile/android-components/blob/d4307a6917913bf5e5776b0b96f1929dd645653c/components/support/utils/src/main/java/mozilla/components/support/utils/Browsers.kt#L125).

Already changed it. 😄

**Review by Amejia481 — APPROVED** [review]

**Comment by Amejia481** [comment]
Great job!!!!

**Pull request merged**